### PR TITLE
feat(tracking): experimental pixel tracking through log files

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Implements an alternative strategy for pixel tracking, suggested by @claudiulodro, by storing temporary log files that are processed through cron instead of processing each on every pixel request.

The cron, executed every minute, looks for an existing log file and processes each stored request. It will, then, delete the processed file and generate a new one using [`tempnam()`](https://www.php.net/manual/en/function.tempnam.php) for a new unique file name.

When a new log file is created, the `np-newsletters-pixel.php` file is also dynamically replaced to account for the changed file name.

This PR also implements a filter to the tracking pixel URL, used by the new strategy to point to the new `np-newsletters-pixel.php`.

Cycling log files with a unique name is the security layer I'm going for since the logs are stored in the uploads directory. I am open to suggestions for alternatives or improvements to this strategy.

Due to its experimental nature, it's currently placed behind the `NEWSPACK_NEWSLETTERS_PIXEL_LOG_PROCESSING` PHP constant and the default/existing approach should continue to work without issues.

### How to test the changes in this Pull Request:

1. Without yet modifying the `NEWSPACK_NEWSLETTERS_PIXEL_LOG_PROCESSING`, send a new newsletter, open it, and confirm the pixel tracking behaves as expected (incremented "Opens" in the newsletters dashboard table)
2. Add `define( 'NEWSPACK_NEWSLETTERS_PIXEL_LOG_PROCESSING', true );` to your `wp-config.php` file
3. Also confirm you have `define( 'NEWSPACK_LOG_LEVEL', 2 );` so we can track the data event being dispatched
4. Send a new newsletter and open it
5. Navigate to your `uploads/2023/10` directory and confirm a `newspack_newsletters_pixel_log_{xxxxxx}` file exists and contains a base64 encoded line
6. Open the email again and confirm the file has now 2 lines
7. Navigate a bit on your site until the cron executes another log processing
8. Confirm the file is replaced and the server logged `[49160][NEWSPACK-DATA-EVENTS][Newspack\Data_Events::dispatch]: Dispatching action "newsletter_opened" via Action Scheduler`
9. Confirm the "Opens" column in the newsletters dashboard table also had its value increment for the sent newsletter

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/tracking/class-pixel.php [hunk 1]
```diff
@@ -207,5 +222,125 @@ public static function render() {
...
+					continue;
+				}
+				// Values must be sanitized as they are stored in the logs without sanitization.
+				$newsletter_id = isset( $data['id'] ) ? intval( $data['id'] ) : 0;
+				$tracking_id   = isset( $data['tid'] ) ? \sanitize_text_field( $data['tid'] ) : 0;
+				$email_address = isset( $item[0] ) ? \sanitize_email( $item[0] ) : '';
+				if ( ! $newsletter_id || ! $tracking_id || ! $email_address ) {
+					continue;
+				}
+				self::track_seen( $newsletter_id, $newsletter_id, $email_address );
```
Comment (adekbadek) [inline includes/tracking/class-pixel.php:293]: Should be:

```suggestion
				self::track_seen( $newsletter_id, $tracking_id, $email_address );
```

File: includes/tracking/class-pixel.php [hunk 2]
```diff
@@ -207,5 +222,125 @@ public static function render() {
...
+	 * @param int    $post_id     ID of the newsletter.
+	 * @param string $tracking_id Tracking ID.
+	 * @param string $email_tag   Email address of the recipient.
+	 *
+	 * @return string
+	 */
+	public static function log_pixel_url( $url, $post_id, $tracking_id, $email_tag ) {
+		return \add_query_arg(
+			[
+				'data' => \base64_encode(
```
Comment (adekbadek) [inline includes/tracking/class-pixel.php:248]: Why is this content base64-encoded?

**Review by adekbadek — CHANGES_REQUESTED** [review]
Found some minor issues.

File: includes/tracking/class-pixel.php (hunk 2 above)
Comment (miguelpeixe) [reply includes/tracking/class-pixel.php]: That's actually not required at all. Updated in 2efd68c70f7218c2ccd77d6edbc5d61c12ceb082 and 669a35886aa61c04f4f35a6c287b29f7137b5d2b for simplicity.

**Comment by miguelpeixe** [comment]
Moved to #1332

**Pull request closed without merging**